pkg/scheduler/core/assignment.go

```diff
@@ -90,7 +90,17 @@ func newAssignState(candidates []*clusterv1alpha1.Cluster, placement *policyv1al
 }
 
 func (as *assignState) buildScheduledClusters() {
-	as.scheduledClusters = as.spec.Clusters
+	validClusterSet := sets.Set[string]{}
+	for _, c := range as.candidates {
+		validClusterSet.Insert(c.Name)
+	}
+	as.scheduledClusters = []workv1alpha2.TargetCluster{}
+	for _, c := range as.spec.Clusters {
+		if !validClusterSet.Has(c.Name) {
```

Comment:
It would be great to have a comment here, like:
```
		// Ignore clusters that are no longer candidates, to ensure we can get real
		// 'assigned' replicas from the previous schedule result. The ignored replicas
		// will be treated as scaled-up replicas that will be assigned to other
		// candidate clusters.
```